docs/languages/csharp/async.md

```diff
@@ -63,60 +149,66 @@ private async void SeeTheDotNets_Click(object sender, RoutedEventArgs e)
+Here's another way to write this a bit more succinctly, using LINQ:
 
-Although many details were glossed over (such as how “borrowing” compute time on a thread pool is coordinated), the important thing to recognize here is that **no thread is 100% dedicated to running the initiated task**. This allows threads in the thread pool of a system to handle a larger volume of work rather than having to wait for I/O to finish.
+```csharp
 
-Although the above may seem like a lot of work to be done, when measured in terms of wall clock time, it’s miniscule compared to the time it takes to do the actual I/O work. Although not at all precise, a potential timeline for such a call would look like this:
-
-0-1————————————————————————————————————————————————–2-3
+public async Task<User> GetUser(int userId)
+{
+    // Code omitted:
+    //
+    // Given a user Id {userId}, retrieves a User object corresponding
+    // to the entry in the database with {userId} as its Id.
+}
 
-*   Time spent from points `0` to `1` is everything up until an async method yields control to its caller.
-*   Time spent from points `1` to `2` is the time spent on I/O.
-*   Finally, time spent from points `2` to `3` is passing control back (and potentially a value) to the async method, at which point it is executing again.
+public static async Task<User[]> GetUsers(IEnumerable<int> userIds)
+{
+    var getUserTasks = userIds.Select(id => GetUser(id));
+    return await Task.WhenAll(getUserTasks);
+}
+```
 
-Tasks are also used outside of the async programming model. They are the foundation of the Task Parallel Library, which supports the parallelization of CPU-bound work via [Data Parallelism](https://msdn.microsoft.com/en-us/library/dd537608%28v=vs.110%29.aspx) and [Task Parallelism](https://msdn.microsoft.com/en-us/library/dd537609%28v=vs.110%29.aspx).
+Although it's less code, take care when mixing LINQ with asynchronous code.  Because LINQ uses deferred (lazy) execution, async calls won't happen immediately as they do in a `foreach()` loop unless you force the generated sequence to iterate with a call to `.ToList()` or `.ToArray()`.
 
 ## Important Info and Advice
 
 Although async programming is relatively straightforward, there are some details to keep in mind which can prevent unexpected behavior.
 
+*  `async` **methods need to have an** `await` **keyword in their body or they will never yield!**
+
+This is important to keep in mind.  If `await` is not used in the body of an `async` method, the C# compiler will generate a warning, but the code will compile and run as if it were a normal method.
```

Comment:
It's also incredibly inefficient. The compiler generates all the code for the state machine, even though it will never get used.